exporters/sender/okhttp/src/main/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpHttpSender.java

```diff
@@ -101,38 +102,40 @@ public void send(
       requestBuilder.post(body);
     }
 
-    client
-        .newCall(requestBuilder.build())
-        .enqueue(
-            new Callback() {
-              @Override
-              public void onFailure(Call call, IOException e) {
-                onError.accept(e);
-              }
-
-              @Override
-              public void onResponse(Call call, okhttp3.Response response) {
-                try (ResponseBody body = response.body()) {
-                  onResponse.accept(
-                      new Response() {
-                        @Override
-                        public int statusCode() {
-                          return response.code();
+    InstrumentationUtil.suppressInstrumentation(
+        () ->
+            client
+                .newCall(requestBuilder.build())
+                .enqueue(
```

Comment:
Doesn't the async stuff throw a curve ball here? Do we need to keep the context until the async `onFailure`/`onResponse` complete?

If not, and it just works, can we get an async test example in place?

Reply:
This is a good point, since we keep the caller's Context across threads in the automatic instrumentation, we'd pass the one with the "suppress_instrumentation" key over to the callback functions onFailure/onResponse. We can probably avoid doing so by wrapping our callbacks in a Callback implementation that restores the original context, the one before adding the "suppress_instrumentation" key.

I think it's worth mentioning that this won't happen to non-OTel usages of OkHttp, since you'd need to manually wrap your okhttp call with `InstrumentationUtil.suppressInstrumentation` for the context key to show up in your callback's functions.

> can we get an async test example in place?
I'm up for adding one but I'm not sure how to make it work in this repo since we don't have bytecode instrumentation in here. What would happen is that the context wouldn't get propagated over to the secondary thread, since that's done by the bytecode instrumentation agent. There's probably an alternative to instrumenting bytecode that we could use for tests that I might've missed though, so I'm open to alternatives!

Reply:
I've added some changes to the `DaemonThreadFactory` class to enable a test that relies on context propagation across threads. Please have a look @breedx-splk, I'm not sure if it's too "hacky" though.